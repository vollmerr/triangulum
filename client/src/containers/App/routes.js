import CrawlForm from '../CrawlForm';
import CrawlGraph from '../CrawlGraph';
import CrawlHistory from '../CrawlHistory';

const routes = {
  form: {
    key: 'form',
    exact: true,
    name: 'New',
    path: '/',
    icon: 'plus',
    component: CrawlForm,
  },
  graph: {
    key: 'graph',
    name: 'Graph',
    path: '/graph',
    icon: 'chart area',
    component: CrawlGraph,
  },
  history: {
    key: 'history',
    name: 'History',
    path: '/history',
    icon: 'history',
    component: CrawlHistory,
  },
};

export default Object.values(routes);
