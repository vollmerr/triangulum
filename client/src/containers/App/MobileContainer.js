import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import {
  Container,
  Icon,
  Menu,
  Responsive,
  Segment,
  Sidebar,
} from 'semantic-ui-react';

import routes from './routes';

const Links = ({ path, onClick }) => (
  routes.map(route => (
    <Menu.Item
      key={route.key}
      icon={route.icon}
      content={route.name}
      as={Link}
      to={route.path}
      active={path === route.path}
      onClick={onClick}
    />
  ))
);

class MobileContainer extends Component {
  state = {}

  closeMenu = () => {
    const { isMenuOpen } = this.state;
    if (isMenuOpen) {
      this.setState({ isMenuOpen: false });
    }
  }

  toggleMenu = () => {
    const { isMenuOpen } = this.state;
    this.setState({ isMenuOpen: !isMenuOpen });
  }

  render() {
    const { children, location } = this.props;
    const { isMenuOpen } = this.state;

    return (
      <Responsive maxWidth={Responsive.onlyMobile.maxWidth} id={'mobile-view'}>
        <Sidebar.Pushable>
          <Sidebar as={Menu} animation='overlay' vertical visible={isMenuOpen} direction={'right'}>
            <Links path={location.pathname} onClick={this.closeMenu} />
          </Sidebar>

          <Sidebar.Pusher
            dimmed={isMenuOpen}
            onClick={this.closeMenu}
          >
            <Segment
              inverted
              textAlign='center'
              style={{ minHeight: 40, padding: '1em 0em', margin: '0 0 1.5em 0' }}
              color={'violet'}
              vertical
            >
              <Container>
                <Menu inverted pointing secondary fixed='top'>
                  <Menu.Item
                    icon={'sitemap'}
                    style={{ transform: 'rotate(0.5turn)' }}
                    as={Link}
                    to={'/'}
                  />
                  <Menu.Item onClick={this.toggleMenu} position='right'>
                    <Icon name='sidebar' />
                  </Menu.Item>
                </Menu>
              </Container>
            </Segment>

            {children}
          </Sidebar.Pusher>
        </Sidebar.Pushable>
      </Responsive>
    );
  }
}

export default withRouter(MobileContainer);
