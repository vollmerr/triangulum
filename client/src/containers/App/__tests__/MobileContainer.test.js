import React from 'react';
import { shallow } from 'enzyme';
import { Menu } from 'semantic-ui-react';

import routes from '../routes';
import { MobileContainer, Links } from '../MobileContainer';

let wrapper;
let instance;
const props = {
  location: {
    pathname: '/test-path',
  },
};
const Child = () => <div>children...</div>;

describe('Links', () => {
  it('should map all the routes as menu items', () => {
    wrapper = shallow(<Links />);
    const items = wrapper.find(Menu.Item);
    expect(items.length).toEqual(routes.length);
  });
});

describe('MobileContainer', () => {
  beforeEach(() => {
    wrapper = shallow((
      <MobileContainer {...props}>
        <Child />
      </MobileContainer>
    ));
    instance = wrapper.instance();
  });

  describe('closeMenu', () => {
    it('should close the menu if it is open', () => {
      wrapper.setState({ isMenuOpen: true });
      instance.closeMenu();
      expect(wrapper.state('isMenuOpen')).toBeFalsy();
    });

    it('should do nothing if the menu is not open', () => {
      wrapper.setState({ isMenuOpen: false });
      instance.closeMenu();
      expect(wrapper.state('isMenuOpen')).toBeFalsy();
    });
  });

  describe('toggleMenu', () => {
    it('should toggle the menu from false to true', () => {
      wrapper.setState({ isMenuOpen: true });
      instance.toggleMenu();
      expect(wrapper.state('isMenuOpen')).toBeFalsy();
    });

    it('should toggle the menu from true to false', () => {
      wrapper.setState({ isMenuOpen: false });
      instance.toggleMenu();
      expect(wrapper.state('isMenuOpen')).toBeTruthy();
    });
  });

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the chidlren', () => {
      expect(wrapper.find(Child).length).toEqual(1);
    });
  });
});
