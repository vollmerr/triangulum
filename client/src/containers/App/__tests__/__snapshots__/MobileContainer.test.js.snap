// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileContainer render should render correctly 1`] = `
<Responsive
  getWidth={[Function]}
  id="mobile-view"
  maxWidth={767}
>
  <SidebarPushable>
    <Sidebar
      animation="overlay"
      as={[Function]}
      direction="right"
      duration={500}
      vertical={true}
    >
      <Links
        onClick={[Function]}
        path="/test-path"
      />
    </Sidebar>
    <SidebarPusher
      onClick={[Function]}
    >
      <Segment
        color="violet"
        inverted={true}
        style={
          Object {
            "margin": "0 0 1.5em 0",
            "minHeight": 40,
            "padding": "1em 0em",
          }
        }
        textAlign="center"
        vertical={true}
      >
        <Container>
          <Menu
            fixed="top"
            inverted={true}
            pointing={true}
            secondary={true}
          >
            <MenuItem
              as={[Function]}
              icon="sitemap"
              style={
                Object {
                  "transform": "rotate(0.5turn)",
                }
              }
              to="/"
            />
            <MenuItem
              onClick={[Function]}
              position="right"
            >
              <Icon
                as="i"
                data-cy="button-menu"
                name="sidebar"
              />
            </MenuItem>
          </Menu>
        </Container>
      </Segment>
      <Child />
    </SidebarPusher>
  </SidebarPushable>
</Responsive>
`;
